### PR TITLE
Log ``worker_client`` event

### PR DESCRIPTION
**Pull request by jrbourbeau**
Using `worker_client` is supported but using them well can require some care. This PR makes sure we log an event when creating a client via `worker_client`, which can be used later when debugging, etc.

### CODE REVIEW

File: distributed/worker_client.py [hunk 1]
```diff
@@ -53,6 +53,14 @@ def worker_client(timeout=None, separate_thread=True):
 
     worker = get_worker()
     client = get_client(timeout=timeout)
+    worker.log_event(
+        worker.address,
+        {
+            "action": "worker-client",
+            "timeout": timeout,
+            "separate_thread": separate_thread,
```
Comment (jrbourbeau) [inline distributed/worker_client.py:61]: Not sure how useful this extra info actually is

**Review by phofl — COMMENTED** [review]
The change itself looks good to me, but it's missing the case that is more critical to me:

```
@delayed
def bla(i):
    arr = da.random.random((500, 500, 100), chunks=(10, 10, 10))
    result = dask.compute(arr.sum())
    return result

if __name__ == "__main__":
    cluster = LocalCluster()
    client = cluster.get_client()
    f = bla(1)
    dask.compute(f)
```

the accidental worker client creation. I think this happens somewhere in ``_get_client`` on the worker

**Comment by jrbourbeau** [comment]
> but it's missing the case that is more critical to me

Ah, yeah, we definitely want to capture that case too. I've updated things to capture both the implicit client creation case via `.compute()` and the explicit `worker_client` case. It's not easy to distinguish between the two cases though, so we're logging the same event in both cases.

**Comment by phofl** [comment]
Could we log an event where you have it now and then a different event in the worker_client context manager?

Subtracting both counters would then tell us if it's intentional or accidental?

**Comment by jrbourbeau** [comment]
Sure, we now have a `worker-get-client` event which is logged in both cases and a `worker-client` event that's only logged in the `worker_client(...)` case

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx

**Pull request merged**